tests/integration/clientv3/metrics_test.go

```diff
@@ -168,11 +170,186 @@ func getHTTPBodyAsLines(t *testing.T, url string) []string {
+		"etcd_disk_defrag_inflight",
+		"etcd_disk_wal_fsync_duration_seconds_bucket",
+		"etcd_disk_wal_fsync_duration_seconds_count",
+		"etcd_disk_wal_fsync_duration_seconds_sum",
+		"etcd_disk_wal_write_bytes_total",
+		"etcd_disk_wal_write_duration_seconds_bucket",
+		"etcd_disk_wal_write_duration_seconds_count",
+		"etcd_disk_wal_write_duration_seconds_sum",
+		"etcd_mvcc_db_open_read_transactions",
+		"etcd_mvcc_db_total_size_in_bytes",
+		"etcd_mvcc_db_total_size_in_use_in_bytes",
+		"etcd_mvcc_delete_total",
+		"etcd_mvcc_hash_duration_seconds_bucket",
+		"etcd_mvcc_hash_duration_seconds_count",
+		"etcd_mvcc_hash_duration_seconds_sum",
+		"etcd_mvcc_hash_rev_duration_seconds_bucket",
+		"etcd_mvcc_hash_rev_duration_seconds_count",
+		"etcd_mvcc_hash_rev_duration_seconds_sum",
+		"etcd_mvcc_put_total",
+		"etcd_mvcc_range_total",
+		"etcd_mvcc_txn_total",
+		"etcd_network_client_grpc_received_bytes_total",
+		"etcd_network_client_grpc_sent_bytes_total",
+		"etcd_network_known_peers",
+		"etcd_server_apply_duration_seconds_bucket",
+		"etcd_server_apply_duration_seconds_count",
+		"etcd_server_apply_duration_seconds_sum",
+		"etcd_server_client_requests_total",
+		"etcd_server_go_version",
+		"etcd_server_has_leader",
+		"etcd_server_health_failures",
+		"etcd_server_health_success",
+		"etcd_server_heartbeat_send_failures_total",
+		"etcd_server_id",
+		"etcd_server_is_leader",
+		"etcd_server_is_learner",
+		"etcd_server_leader_changes_seen_total",
+		"etcd_server_learner_promote_successes",
+		"etcd_server_proposals_applied_total",
+		"etcd_server_proposals_committed_total",
+		"etcd_server_proposals_failed_total",
+		"etcd_server_proposals_pending",
+		"etcd_server_quota_backend_bytes",
+		"etcd_server_read_indexes_failed_total",
+		"etcd_server_slow_apply_total",
+		"etcd_server_slow_read_indexes_total",
+		"etcd_server_snapshot_apply_in_progress_total",
+		"etcd_server_version",
+		"etcd_snap_db_fsync_duration_seconds_bucket",
+		"etcd_snap_db_fsync_duration_seconds_count",
+		"etcd_snap_db_fsync_duration_seconds_sum",
+		"etcd_snap_db_save_total_duration_seconds_bucket",
+		"etcd_snap_db_save_total_duration_seconds_count",
+		"etcd_snap_db_save_total_duration_seconds_sum",
+		"etcd_snap_fsync_duration_seconds_bucket",
+		"etcd_snap_fsync_duration_seconds_count",
+		"etcd_snap_fsync_duration_seconds_sum",
+		"grpc_client_handled_total",
+		"grpc_client_msg_received_total",
+		"grpc_client_msg_sent_total",
+		"grpc_client_started_total",
+		"grpc_server_handled_total",
+		"grpc_server_msg_received_total",
+		"grpc_server_msg_sent_total",
+		"grpc_server_started_total",
+	}
+
+	// Get the list of generated metrics
+	generatedMetrics := getMetricsList(t, url)
+	for _, metric := range expectedMetrics {
+		if !slices.Contains(generatedMetrics, metric) {
+			t.Errorf("Expected metric %s not found in generated metrics", metric)
+		}
+	}
+}
+
+func getMetricsList(t *testing.T, url string) []string {
```

Comment:
FYI "github.com/prometheus/common/expfmt" package has a parser for metrics. Example usage https://github.com/kubernetes-sigs/metrics-server/blob/0dcee07ea614e8354ec740dfc3f528a572cacec4/test/e2e_test.go#L315-L329

Reply:
@serathius I will update the implementation of this test as suggested in the other PR when i rebase!